chrome/script.browserify.js

```diff
@@ -86,15 +96,77 @@ function view() {
   ]);
 }
 
+function filterLogins(e) {
+  // remove executed search from input field
+  if (!fillOnSubmit && e.target.value.indexOf(domain) === 0) {
+    e.target.value = e.target.value.substr(domain.length);
+  }
+
+  // use fuzzy search to filter results
+  var filter = e.target.value.trim().split(/[\s\/]+/);
+  if (filter.length > 0) {
+    logins = resultLogins.slice(0);
+    filter.forEach(function(word) {
+      if (word.length > 0) {
+        var refine = [];
+        FuzzySort.go(word, logins, {allowTypo: false}).forEach(function(result) {
+          refine.push(result.target);
+        });
+        logins = refine.slice(0);
+      }
+    });
+
+    // fill login forms on submit rather than initiating a search
+    fillOnSubmit = logins.length > 0;
+  } else {
+    // reset the result list if the filter is empty
+    logins = resultLogins.slice(0);
+  }
+
+  // redraw the list
+  m.redraw();
+
+  // show / hide the filter hint
+  showFilterHint(logins && logins.length);
+}
+
+function searchKeyHandler(e) {
+  // switch to search mode if '\' is pressed and no filter text has been entered
+  if (e.code == "Backspace" && logins && logins.length > 0 && (!e.target.value.length || e.target.value == domain)) {
+    e.preventDefault();
+    logins = resultLogins = [];
+    e.target.value = '';
+    showFilterHint(false);
+  }
+}
+
+function showFilterHint(show=true) {
+  var filterHint = document.getElementById("filter-search");
+  var searchField = document.getElementById("search-field");
+  if (show) {
+    filterHint.style.display = "block";
+    searchField.setAttribute("placeholder", "Refine search...");
+  } else {
+    filterHint.style.display = "none";
+    searchField.setAttribute("placeholder", "Search passwords...");
+  }
+}
+
 function submitSearchForm(e) {
   e.preventDefault();
 
-  // don't search without input.
-  if (!this.s.value.length) {
-    return;
-  }
+  if (fillOnSubmit && logins && logins.length > 0) {
```

Comment:
I think `&& logins` is not needed anymore since it can never be null

Reply:
Done.